### PR TITLE
tests.cc-wrapper: skip known-broken sanitizer cases

### PR DESCRIPTION
**Pull request by Ralith**
###### Motivation for this change
Fixes overly-broad testing due to #41065. Let me know if there's a better way to test for libc++ being in use.

CC @Ericson2314 @LnL7 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/build-support/cc-wrapper/default.nix [hunk 1]
```diff
@@ -90,7 +90,7 @@ stdenv.mkDerivation {
     # Binutils, and Apple's "cctools"; "bintools" as an attempt to find an
     # unused middle-ground name that evokes both.
     inherit bintools;
-    inherit libc nativeTools nativeLibc nativePrefix isGNU isClang default_cxx_stdlib_compile;
+    inherit libc nativeTools nativeLibc nativePrefix isGNU isClang isLibCxx default_cxx_stdlib_compile;
```
Comment (LnL7) [inline pkgs/build-support/cc-wrapper/default.nix]: What about introducing libcxx similar to libc we have now? I think we want to add that in the future anyway to get rid of the `default_cxx_stdlib_compile` hack.

Comment (Ralith) [reply pkgs/build-support/cc-wrapper/default.nix]: Is there documentation anywhere of what exactly the `libc` we have now *is*?

**Comment by Ericson2314** [comment]
Hmm so this doesn't affect the libstdc++ ones? That's very very odd, as the sanitizer runtimes don't come from either.

**Comment by Ralith** [comment]
Whoops, looks like that's an issue to.

Is there any way to run all the affected tests, so I don't have to keep making slightly inaccurate guesses about what to try?

**Comment by Ericson2314** [comment]
Sadly I don't know of an easy way to replicate what hydra does, as conceptually simple as it is.

**Comment by matthewbauer** [comment]
This is still being run:

https://hydra.nixos.org/build/75406498/nixlog/1

Is thaat conditional correct?

**Comment by matthewbauer** [comment]
@Ralith Do you know what's going on with this latest error:

https://hydra.nixos.org/build/75406498/nixlog/1

```
ld: file not found: /nix/store/wdiibxx7221iiv8cdgdafgbmkvb58bw9-clang-wrapper-6.0.0/resource-root/lib/darwin/libclang_rt.asan_osx_dynamic.dylib
clang-6.0: error: linker command failed with exit code 1 (use -v to see invocation)
builder for '/nix/store/ih843k6q8xv3xbakp7r370377z18wnci-cc-wrapper-test.drv' failed with exit code 1
```

**Comment by orivej** [comment]
I have verified that sanitizers did not work on Darwin neither when #39743 was merged in 713d580d12b [1], nor directly before that in ae995625d8e [2]. Therefore it is reasonable to disable sanitizers tests on Darwin for now.

[1] With `tests.cc-wrapper-clang-6` the error is `ld: file not found: /nix/store/2qfdihv2x3m6da4bpv9jpi1qwlnzkns6-clang-wrapper-6.0.0/resource-root/lib/darwin/libclang_rt.asan_osx_dynamic.dylib`. File tree:
```
/nix/store/2qfdihv2x3m6da4bpv9jpi1qwlnzkns6-clang-wrapper-6.0.0/resource-root/lib
└── darwin
    ├── libclang_rt.10.4.a
    ├── libclang_rt.cc_kext.a
    └── libclang_rt.osx.a
```

With `tests.cc-wrapper-gcc`, the error is the the same as in [2].

[2] The error is:
```
/nix/store/3c3vrabg4yy87y4qx8zsy5x2vdg132j5-sanitizers.c:1:10: fatal error: 'sanitizer/asan_interface.h' file not found
#include <sanitizer/asan_interface.h>
```

**Comment by orivej** [comment]
Disabled sanitizer tests on Darwin in 07ebb8bb792be16de33596e2cea120eb72b2ec77.

**Comment by Ralith** [comment]
Thanks for handling that! I wonder why the asan stuff is missing. Maybe some option we need to patch to the compiler-rt build, or perhaps one of our darwin-specific patches or options is disabling it.

**Pull request merged**